repos/system_upgrade/el7toel8/actors/checksaphana/libraries/checksaphana.py

```diff
@@ -0,0 +1,160 @@
+from leapp.libraries.stdlib import api
+from leapp.libraries.common.config import architecture
+from leapp.models import SapHanaInfo
+from leapp import reporting
+
+
+# SAP HANA Compatibility
+
+SAP_HANA_MINIMAL_MAJOR_VERSION = 2
+SAP_HANA_MINIMAL_SP = 5
+SAP_HANA_MINIMAL_REV = 52
+SAP_HANA_MINIMAL_VERSION_STRING = 'HANA 2.0 SPS05 rev 52'
+
+
+def running_check(info):
+    if info.running:
+        reporting.create_report([
+            reporting.Title('Found running SAP HANA instances'),
+            reporting.Summary(
+                'In order to perform a system upgrade it is necessary that all instances of SAP Hana are stopped.'
+                'Please shutdown all SAP Hana instances before you continue with the upgrade.'
+            ),
+            reporting.Severity(reporting.Severity.HIGH),
+            reporting.Tags([reporting.Tags.SANITY]),
+            reporting.Flags([reporting.Flags.INHIBITOR]),
+            reporting.Audience('sysadmin')
+        ])
+
+
+def _add_hana_details(target, instance):
+    target.setdefault(instance.name, {'numbers': set(), 'path': instance.path, 'admin': instance.admin})
+    target[instance.name]['numbers'].add(instance.instance_number)
+
+
+def _create_detected_instances_list(details):
+    result = []
+    for name, meta in details.items():
+        result.append(('Name: {name}\n'
+                       '  Instances: {instances}\n'
+                       '  Admin: {admin}\n'
+                       '  Path: {path}').format(name=name,
+                                                instances=', '.join(meta['numbers']),
+                                                admin=meta['admin'],
+                                                path=meta['path']))
+    if result:
+        return '- {}'.format('\n- '.join(result))
+    return ''
+
+
+def version1_check(info):
+    found = {}
+    for instance in info.instances:
+        if instance.manifest.get('release', None) == '1.00':
+            _add_hana_details(found, instance)
+
+    if found:
+        detected = _create_detected_instances_list(found)
+        reporting.create_report([
+            reporting.Title('Found outdated SAP HANA 1'),
+            reporting.Summary(
+                ('SAP Hana 1.00 is not supported on the version of RHEL you are'
+                 ' upgrading to. In order to upgrade RHEL, you will have to'
+                 ' upgrade SAP Hana first to the latest version of SAP Hana 2.00.\n\n'
+                 'The following instances have been detected to be running version 1.00:\n'
+                 '{}'.format(detected))
```

Comment:
nitpick. do we have a doc, ..., so we could attach the link for info about supported version of sap?

Reply:
https://launchpad.support.sap.com/#/notes/2235581

Reply:
I am not able to access the content. And ideally we should point just to redhat official documentation - not outside doc..